### ISSUE TITLE
NVDA doesn't read aria-label and aria-describedby for field focused by anchor link

### ISSUE THREAD

**Issue by eumis** [nvaccess/nvda#6209]
Hi, i found issue for nvda with firefox.
If we have anchor link to some text field, by clicking on that link nvda ignores aria-label and aria-described attributes.
Piece of code for example:

`
<a href='#textInput'>Link to input</a>
<input type="text" id='textInput' aria-label='some label' aria-describedby="description" />
<span id='description'>some description</span>
`

**Comment by SomethingNew71** [comment]
+1 I am also having this problem

**Comment by ccruz2886** [comment]
It is happening to me as well... It is ignoring `aria-describedby` while I'm trying to read an anchor element. In my case doesn't matter the browser.

**Comment by 36degrees** [comment]
Still seeing this behaviour in NVDA 2018.2 with Firefox 60.2.1esr on Windows 7 Professional SP1.

In our testing, the label is also ignored.

Given the following HTML snippet:

```
<a href="#textinput">Link to input</a>

<label id="textinput-label" for="textinput">Label for text input</label>
<span class="error" id="textinput-error">Error for text input</span>
<input id="textinput" type="text" aria-describedby="textinput-error" value="Current value">
```

The **expected behaviour** would mirror NVDA's existing behaviour when focussing an input by tabbing to it:

1. Click the link 'Link to text input'
2. The input is focussed
3. NVDA announces "Label for text input, edit has autocomplete, error for text input, selected, current value"

The **actual behaviour** is that NVDA omits both the label and the error message (associated using aria-describedby):

1. Click the link 'Link to text input'
2. The input is focussed
3. NVDA announces "Edit has autocomplete, current value"

In our testing this same problem exists when linking to all types of inputs, including textareas, selects, radio buttons and checkboxes.

_This is part of a wider piece of work exploring how we should link to inputs of different types from validation summaries. If it's helpful, you can also see our [full test case](https://36degrees.github.io/linking-to-form-inputs) at and the [results of our testing different options in assistive technologies](https://36degrees.github.io/linking-to-form-inputs/results/)._

**Comment by 36degrees** [comment]
We have found that manually [managing the focus using JavaScript](https://github.com/alphagov/govuk-frontend/blob/e7cfd476c6ed05c43bd1b37b326e699db3d8c8a0/src/components/error-summary/error-summary.js#L33-L85) (preventing the default event handler on the link, and calling `focus()` on the target input) [results in NVDA announcing the label, legend and description 'correctly'](https://github.com/alphagov/govuk-frontend/pull/1056#issuecomment-438625484)

(We're doing this for other reasons, so that we can ensure the label or legend remain within the viewport when targeting the input – this just happens to be a positive side-effect)

**Comment by Adriani90** [comment]
This seems to be very inconsistent between browsers. I have tested both code pieces. Both of them work correctly only in Google Chrome and Edge, there is no issue.

In Firefox 65.0.2 and last NVDA Alpha, with the first code piece in the description, NVDA reports the label but not the description. With the second code piece in the comment above, NVDA does not report the label neither the description. @jcsteh is there a bug filled against Firefox regarding this? Or is this a bug in NVDA?


Additional information: In IE neither the label nor the descriiption are reported.

**Comment by Adriani90** [comment]
This seems to work as expected now with NVDA last alpha version and firefox 111 as well as Chrome and Edge 112.

You can test with the last NVDA alpha from this link:
https://www.nvaccess.org/files/nvda/snapshots/

I am closing as works for me.

Please comment on this with updated steps to reproduce if you are still having this issue and we can reopen this one.
